{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 102,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 103,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "import numpy as np"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 104,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "from glob import glob"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 105,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "import itertools"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 106,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "import pygsheets as pyg"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 107,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "import time"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### some pandas opts"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 108,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "pd.options.display.max_columns = 100"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 109,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "pd.options.display.max_rows = 500"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### authorize pyg and set retry limit"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 110,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "gc = pyg.authorize(outh_file=\"./client_secret_986841527205-ndf6fbsfmrhjinnofm47j6olvmm11smn.apps.googleusercontent.com.json\",\n",
    "                   outh_nonlocal=True,\n",
    "                   retries=5)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### title for the google sheet and xls"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 111,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "sheet_name = \"BY-nup133-day70\""
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### xls output directory"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 112,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "xls_out_dir = \"/home/dizak/Pulpit/BIONAS/G148/SNPs_calling/BY-nup133-day70/\""
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### where are the files?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 113,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[0m\u001b[01;32m16 documents from consensus.geneious\u001b[0m*\r\n",
      "\u001b[01;32m16 documents from WT-day70 to WT-zero point.geneious\u001b[0m*\r\n",
      "\u001b[34;42mBY-day0\u001b[0m/\r\n",
      "\u001b[34;42mBY-nup133-day70\u001b[0m/\r\n",
      "\u001b[34;42mBY-WT-day70\u001b[0m/\r\n",
      "\u001b[01;32mC2_merged_dfs.csv\u001b[0m*\r\n",
      "\u001b[34;42mconsensus\u001b[0m/\r\n",
      "\u001b[34;42mmapped_contigs\u001b[0m/\r\n",
      "\u001b[34;42mmapped_reads\u001b[0m/\r\n",
      "\u001b[01;32mnup mapped to wt cons.geneious\u001b[0m*\r\n",
      "\u001b[34;42mW303-cog7-day42\u001b[0m/\r\n",
      "\u001b[34;42mW303-nup133-day42\u001b[0m/\r\n",
      "\u001b[34;42mW303-WT-day42\u001b[0m/\r\n",
      "\u001b[34;42mWT-day70\u001b[0m/\r\n",
      "\u001b[01;32mWT-day70.xlsx\u001b[0m*\r\n"
     ]
    }
   ],
   "source": [
    "%ls /home/dizak/Pulpit/BIONAS/G148/SNPs_calling/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 114,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "sampling_levels = [\"N1\", \"N2\", \"N3\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 115,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "strain_gene_day_N1_files = glob(\"/home/dizak/Pulpit/BIONAS/G148/SNPs_calling/BY-nup133-day70/N1/*csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 116,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "strain_gene_day_N2_files = glob(\"/home/dizak/Pulpit/BIONAS/G148/SNPs_calling/BY-nup133-day70/N2/*csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 117,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "strain_gene_day_N3_files = glob(\"/home/dizak/Pulpit/BIONAS/G148/SNPs_calling/BY-nup133-day70/N3/*csv\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### get non-redundant list of genes in the inputfiles"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 118,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "def find_flat_value(inputfiles_list,\n",
    "                    col_name = \"CDS\"):\n",
    "    \"\"\"\n",
    "    Get flat list of desired values from list of CSV files.\n",
    "    \n",
    "    Parameters\n",
    "    -------\n",
    "    inputfiles_list: list of str\n",
    "        List of input CSV files.\n",
    "    col_name: str\n",
    "        Desired column name in the input CSV file.\n",
    "    \n",
    "    Returns\n",
    "    -------\n",
    "    list of desired values.\n",
    "    \"\"\"\n",
    "    values_list = []\n",
    "    for i in inputfiles_list:\n",
    "        df = pd.read_csv(i)\n",
    "        if len(df) == 0:\n",
    "            pass\n",
    "        elif col_name not in df.columns:\n",
    "            pass\n",
    "        else:\n",
    "            values_list.append(df[col_name].dropna().drop_duplicates().tolist())\n",
    "    return list(itertools.chain.from_iterable(values_list))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### get values from input files by key"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 119,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "def find_by_key(inputfiles_list,\n",
    "                key):\n",
    "    \"\"\"\n",
    "    Get pandas.DataFrame selected by a given key from list of CSV files.\n",
    "    \n",
    "    Parameters\n",
    "    -------\n",
    "    inputfiles_list: list of str\n",
    "        List of input CSV files.\n",
    "    key: str, int, float, bool\n",
    "        Key used as query against rows in the CSV files.\n",
    "    value_col: str\n",
    "        Column name which holds values to be returned.\n",
    "    \n",
    "    Returns\n",
    "    -------\n",
    "    dict of lists of desired values if pandas.Dataframe not empty\n",
    "    None if pandas.DataFrame empty\n",
    "    \"\"\"\n",
    "    values_list = []\n",
    "    for i in inputfiles_list:\n",
    "        filename = \"\".join(i.split(\"/\")[-1].split(\".\")[:-1])\n",
    "        df = pd.read_csv(i)\n",
    "        if len(df) == 0:\n",
    "            pass\n",
    "        else:\n",
    "            if isinstance(key, str) == True:\n",
    "                df_dtype_sel = df.select_dtypes(include=[\"object\"])\n",
    "            elif isinstance(key, int) == True:\n",
    "                df_dtype_sel = df.select_dtypes(include=[\"int\"])\n",
    "            elif isinstance(key, float) == True:\n",
    "                df_dtype_sel = df.select_dtypes(include=[\"float\"])\n",
    "            elif isinstance(key, bool) == True:\n",
    "                df_dtype_sel = df.select_dtypes(include=[\"bool\"])\n",
    "            else:\n",
    "                raise ValueError(\"key must str, int, float or bool dtype\")\n",
    "            for col in df_dtype_sel.columns:\n",
    "                df_sel = df[df_dtype_sel[col] == key]\n",
    "                if len(df_sel) > 0:\n",
    "                    return {\"dataframe\": df_sel,\n",
    "                            \"filename\": filename}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### get whole set of pandas.DataFrames selections in one dict"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 120,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "def get_dfs_set(key_list,\n",
    "                files_list,\n",
    "                vals=[\"Minimum\",\n",
    "                      \"Maximum\",\n",
    "                      \"Change\"],\n",
    "                df_key=\"dataframe\",\n",
    "                key_index_name=\"Gene\",\n",
    "                smpl_index_name=\"Sample\",\n",
    "                row_index_name=\"Number\",\n",
    "                index_by_key=True,\n",
    "                smpl_index_val=None,):\n",
    "    \"\"\"\n",
    "    Get desired values in pandas.Dataframe gathered in dict by\n",
    "    the list of keys.\n",
    "    \n",
    "    Parameters\n",
    "    -------\n",
    "    key_list: list, tuple\n",
    "        List of keys by which pandas.Dataframes are\n",
    "        initially selected.\n",
    "    files_list: list, tuple\n",
    "        List of input CSV files.\n",
    "    vals: list of str, default: [\"Minimum\", \n",
    "                                 \"Maximum\",\n",
    "                                 \"Change\"]\n",
    "        Columns names holding data in pandas.DataFrames\n",
    "        selected.\n",
    "    df_key: str, default: <\"dataframe\">\n",
    "        Key for generic pandas.DataFrame selection for\n",
    "        SNPs-sheets_merge.find_by_key function.\n",
    "    key_index_name: str, default: <\"Gene\">\n",
    "        Name for index of selection key.\n",
    "    smpl_index_name: str, default: <\"Sample\">\n",
    "        Name for index of sample.\n",
    "    row_index_name: str, default: <\"Number\">\n",
    "        Name for numeric row index.\n",
    "    index_by_key: bool, default: True\n",
    "        Enables multiindexing.\n",
    "    smpl_index_val: str, default: <None>\n",
    "        Adds sample name to multiindexing if not <None>\n",
    "    \"\"\"\n",
    "    out_dict = {}\n",
    "    for i in key_list:\n",
    "        key_vals = find_by_key(files_list,\n",
    "                               key=i)[df_key]\n",
    "        out_dict[i] = key_vals[vals]\n",
    "    if index_by_key is True:\n",
    "        for i in out_dict:\n",
    "            key_index = [i] * len(out_dict[i])\n",
    "            if smpl_index_val is not None:\n",
    "                smpl_index = [smpl_index_val] * len(out_dict[i])\n",
    "                tpls = list(zip(*[key_index,\n",
    "                                  smpl_index,\n",
    "                                  out_dict[i].index]))\n",
    "                mindex = pd.MultiIndex.from_tuples(tpls,\n",
    "                                                   names=[key_index_name,\n",
    "                                                          smpl_index_name,\n",
    "                                                          row_index_name])\n",
    "            else:\n",
    "                tpls = list(zip(*[key_index,\n",
    "                                  out_dict[i].index]))\n",
    "                mindex = pd.MultiIndex.from_tuples(tpls,\n",
    "                                                   names=[key_index_name,\n",
    "                                                          row_index_name])\n",
    "            out_dict[i].index = mindex\n",
    "    return out_dict"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### merge any given number of dfs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 121,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "def merge_dfs(dfs,\n",
    "              sort_cols=[\"Minimum\",\n",
    "                         \"Maximum\"],\n",
    "              reconstr_index=True):\n",
    "    \"\"\"\n",
    "    Merge any number of pandas.DataFrame into one.\n",
    "    Indexes must be identical in all the pandas.DataFrames.\n",
    "    \n",
    "    Parameters\n",
    "    -------\n",
    "    dfs: list\n",
    "        list of pandas.DataFrames to merge.\n",
    "    sort_cols: list, None\n",
    "        list of col names to sort the final\n",
    "        pandas.DataFrame by. No sorting if None.\n",
    "    \"\"\"\n",
    "    for x in [set(i.index) for i in dfs]:\n",
    "        assert len(x) == 1, \"Indices are not homogenic.\"\n",
    "    new_index = list(x)\n",
    "    df = reduce(lambda df1, df2: pd.merge(left=df1,\n",
    "                                          right=df2,\n",
    "                                          how=\"outer\"),\n",
    "                dfs)\n",
    "    if sort_cols is not None:\n",
    "        df.sort(columns=sort_cols)\n",
    "    if reconstr_index is True:\n",
    "        df.index = len(df) * new_index\n",
    "    return df"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### let's find out which CDS are present in all the files one by one"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 122,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "strain_gene_day_N1_CDSs = find_flat_value(strain_gene_day_N1_files)\n",
    "strain_gene_day_N2_CDSs = find_flat_value(strain_gene_day_N2_files)\n",
    "strain_gene_day_N3_CDSs = find_flat_value(strain_gene_day_N3_files)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### let's gather some info about each of the CDS from each sample"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 123,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "des_cols = [\"Change\",\n",
    "            \"Protein Effect\",\n",
    "            \"Polymorphism Type\",\n",
    "            \"Variant Frequency\",\n",
    "            \"Average Quality\",\n",
    "            \"Minimum\",\n",
    "            \"Maximum\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 124,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "strain_gene_day_N1_dfs = get_dfs_set(strain_gene_day_N1_CDSs,\n",
    "                                     strain_gene_day_N1_files,\n",
    "                                     vals=des_cols,\n",
    "                                     smpl_index_val=sampling_levels[0])\n",
    "strain_gene_day_N2_dfs = get_dfs_set(strain_gene_day_N2_CDSs,\n",
    "                                     strain_gene_day_N2_files,\n",
    "                                     vals=des_cols,\n",
    "                                     smpl_index_val=sampling_levels[1])\n",
    "strain_gene_day_N3_dfs = get_dfs_set(strain_gene_day_N3_CDSs,\n",
    "                                     strain_gene_day_N3_files,\n",
    "                                     vals=des_cols,\n",
    "                                     smpl_index_val=sampling_levels[2])"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### let's unwind them all from this dict into lists"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 125,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "N1_dfs = [strain_gene_day_N1_dfs[i] for i in strain_gene_day_N1_dfs.keys()]\n",
    "N2_dfs = [strain_gene_day_N2_dfs[i] for i in strain_gene_day_N2_dfs.keys()]\n",
    "N3_dfs = [strain_gene_day_N3_dfs[i] for i in strain_gene_day_N3_dfs.keys()]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### let's merge them to see changes between samples.\n",
    "#### rememeber now cannot use the merge_dfs function since multindex is NOT homogenic"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 126,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "N1_df = reduce(lambda df1, df2: pd.concat([df1, df2]),\n",
    "                N1_dfs)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 127,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "N2_df = reduce(lambda df1, df2: pd.concat([df1, df2]),\n",
    "               N2_dfs)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 128,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "N3_df = reduce(lambda df1, df2: pd.concat([df1, df2]),\n",
    "               N3_dfs)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 129,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "N1_N2_N3_df = reduce(lambda df1, df2: pd.concat([df1, df2]),\n",
    "                     [N1_df, N2_df, N3_df])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 130,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th>Change</th>\n",
       "      <th>Protein Effect</th>\n",
       "      <th>Polymorphism Type</th>\n",
       "      <th>Variant Frequency</th>\n",
       "      <th>Average Quality</th>\n",
       "      <th>Minimum</th>\n",
       "      <th>Maximum</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Gene</th>\n",
       "      <th>Sample</th>\n",
       "      <th>Number</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>MTL1 CDS</th>\n",
       "      <th>N1</th>\n",
       "      <th>0</th>\n",
       "      <td>M -&gt; C</td>\n",
       "      <td>None</td>\n",
       "      <td>SNP</td>\n",
       "      <td>58.0%</td>\n",
       "      <td>26</td>\n",
       "      <td>530038</td>\n",
       "      <td>530038</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"3\" valign=\"top\">DAN4 CDS</th>\n",
       "      <th rowspan=\"3\" valign=\"top\">N1</th>\n",
       "      <th>4</th>\n",
       "      <td>T -&gt; G</td>\n",
       "      <td>Substitution</td>\n",
       "      <td>SNP (transversion)</td>\n",
       "      <td>54.3%</td>\n",
       "      <td>15</td>\n",
       "      <td>715015</td>\n",
       "      <td>715015</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>T -&gt; G</td>\n",
       "      <td>Substitution</td>\n",
       "      <td>SNP (transversion)</td>\n",
       "      <td>63.8%</td>\n",
       "      <td>17</td>\n",
       "      <td>715003</td>\n",
       "      <td>715003</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>C -&gt; G</td>\n",
       "      <td>Substitution</td>\n",
       "      <td>SNP (transversion)</td>\n",
       "      <td>59.1%</td>\n",
       "      <td>16</td>\n",
       "      <td>714979</td>\n",
       "      <td>714979</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"10\" valign=\"top\">SWI1 CDS</th>\n",
       "      <th rowspan=\"10\" valign=\"top\">N1</th>\n",
       "      <th>8</th>\n",
       "      <td>W -&gt; A</td>\n",
       "      <td>Truncation</td>\n",
       "      <td>SNP</td>\n",
       "      <td>97.0%</td>\n",
       "      <td>32</td>\n",
       "      <td>521046</td>\n",
       "      <td>521046</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>M -&gt; A</td>\n",
       "      <td>Truncation</td>\n",
       "      <td>SNP</td>\n",
       "      <td>97.1%</td>\n",
       "      <td>33</td>\n",
       "      <td>521039</td>\n",
       "      <td>521039</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>M -&gt; A</td>\n",
       "      <td>Truncation</td>\n",
       "      <td>SNP</td>\n",
       "      <td>100.0%</td>\n",
       "      <td>34</td>\n",
       "      <td>521036</td>\n",
       "      <td>521036</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>W -&gt; A</td>\n",
       "      <td>Truncation</td>\n",
       "      <td>SNP</td>\n",
       "      <td>99.0%</td>\n",
       "      <td>34</td>\n",
       "      <td>521034</td>\n",
       "      <td>521034</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>W -&gt; A</td>\n",
       "      <td>Truncation</td>\n",
       "      <td>SNP</td>\n",
       "      <td>99.0%</td>\n",
       "      <td>32</td>\n",
       "      <td>521031</td>\n",
       "      <td>521031</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>W -&gt; A</td>\n",
       "      <td>Truncation</td>\n",
       "      <td>SNP</td>\n",
       "      <td>99.0%</td>\n",
       "      <td>32</td>\n",
       "      <td>521027</td>\n",
       "      <td>521027</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>W -&gt; A</td>\n",
       "      <td>Substitution</td>\n",
       "      <td>SNP</td>\n",
       "      <td>97.0%</td>\n",
       "      <td>32</td>\n",
       "      <td>521024</td>\n",
       "      <td>521024</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>MK -&gt; AT</td>\n",
       "      <td>Substitution</td>\n",
       "      <td>Substitution</td>\n",
       "      <td>96.9%</td>\n",
       "      <td>31</td>\n",
       "      <td>521018</td>\n",
       "      <td>521019</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>WW -&gt; TT</td>\n",
       "      <td>Substitution</td>\n",
       "      <td>Substitution</td>\n",
       "      <td>98.0%</td>\n",
       "      <td>31</td>\n",
       "      <td>521014</td>\n",
       "      <td>521015</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>SAW -&gt; GAT</td>\n",
       "      <td>Substitution</td>\n",
       "      <td>Substitution</td>\n",
       "      <td>92.6% -&gt; 92.8%</td>\n",
       "      <td>29</td>\n",
       "      <td>521008</td>\n",
       "      <td>521010</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"4\" valign=\"top\">FLO11 CDS</th>\n",
       "      <th rowspan=\"4\" valign=\"top\">N1</th>\n",
       "      <th>0</th>\n",
       "      <td>C -&gt; G</td>\n",
       "      <td>Substitution</td>\n",
       "      <td>SNP (transversion)</td>\n",
       "      <td>53.8%</td>\n",
       "      <td>12</td>\n",
       "      <td>392337</td>\n",
       "      <td>392337</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>C -&gt; G</td>\n",
       "      <td>Substitution</td>\n",
       "      <td>SNP (transversion)</td>\n",
       "      <td>50.9%</td>\n",
       "      <td>12</td>\n",
       "      <td>392322</td>\n",
       "      <td>392322</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>G -&gt; T</td>\n",
       "      <td>None</td>\n",
       "      <td>SNP (transversion)</td>\n",
       "      <td>53.7%</td>\n",
       "      <td>21</td>\n",
       "      <td>392087</td>\n",
       "      <td>392087</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>G -&gt; A</td>\n",
       "      <td>Substitution</td>\n",
       "      <td>SNP (transition)</td>\n",
       "      <td>66.4%</td>\n",
       "      <td>28</td>\n",
       "      <td>391197</td>\n",
       "      <td>391197</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"5\" valign=\"top\">hypothetical protein CDS</th>\n",
       "      <th rowspan=\"5\" valign=\"top\">N1</th>\n",
       "      <th>6</th>\n",
       "      <td>T -&gt; G</td>\n",
       "      <td>None</td>\n",
       "      <td>SNP (transversion)</td>\n",
       "      <td>73.3%</td>\n",
       "      <td>26</td>\n",
       "      <td>475850</td>\n",
       "      <td>475850</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>CA -&gt; TG</td>\n",
       "      <td>Substitution</td>\n",
       "      <td>Substitution</td>\n",
       "      <td>59.6%</td>\n",
       "      <td>24</td>\n",
       "      <td>475842</td>\n",
       "      <td>475843</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>AACA -&gt; TCAT</td>\n",
       "      <td>Substitution</td>\n",
       "      <td>Substitution</td>\n",
       "      <td>59.6%</td>\n",
       "      <td>24</td>\n",
       "      <td>475837</td>\n",
       "      <td>475840</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>GT -&gt; CA</td>\n",
       "      <td>Substitution</td>\n",
       "      <td>Substitution</td>\n",
       "      <td>58.3%</td>\n",
       "      <td>24</td>\n",
       "      <td>475834</td>\n",
       "      <td>475835</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>CTAA -&gt; ATTT</td>\n",
       "      <td>Substitution</td>\n",
       "      <td>Substitution</td>\n",
       "      <td>50.0% -&gt; 52.2%</td>\n",
       "      <td>23</td>\n",
       "      <td>475827</td>\n",
       "      <td>475830</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"13\" valign=\"top\">FLO9 CDS</th>\n",
       "      <th rowspan=\"13\" valign=\"top\">N1</th>\n",
       "      <th>2</th>\n",
       "      <td>G -&gt; A</td>\n",
       "      <td>None</td>\n",
       "      <td>SNP (transition)</td>\n",
       "      <td>97.6%</td>\n",
       "      <td>24</td>\n",
       "      <td>26497</td>\n",
       "      <td>26497</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>C -&gt; T</td>\n",
       "      <td>Substitution</td>\n",
       "      <td>SNP (transition)</td>\n",
       "      <td>92.1%</td>\n",
       "      <td>22</td>\n",
       "      <td>26492</td>\n",
       "      <td>26492</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>T -&gt; A</td>\n",
       "      <td>None</td>\n",
       "      <td>SNP (transversion)</td>\n",
       "      <td>97.1%</td>\n",
       "      <td>24</td>\n",
       "      <td>26479</td>\n",
       "      <td>26479</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>G -&gt; A</td>\n",
       "      <td>None</td>\n",
       "      <td>SNP (transition)</td>\n",
       "      <td>97.1%</td>\n",
       "      <td>22</td>\n",
       "      <td>26476</td>\n",
       "      <td>26476</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>AG -&gt; GC</td>\n",
       "      <td>Substitution</td>\n",
       "      <td>Substitution</td>\n",
       "      <td>96.9% -&gt; 97.0%</td>\n",
       "      <td>23</td>\n",
       "      <td>26473</td>\n",
       "      <td>26474</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>GCA -&gt; ATG</td>\n",
       "      <td>Substitution</td>\n",
       "      <td>Substitution</td>\n",
       "      <td>96.8%</td>\n",
       "      <td>24</td>\n",
       "      <td>26468</td>\n",
       "      <td>26470</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>C -&gt; T</td>\n",
       "      <td>Substitution</td>\n",
       "      <td>SNP (transition)</td>\n",
       "      <td>100.0%</td>\n",
       "      <td>23</td>\n",
       "      <td>26464</td>\n",
       "      <td>26464</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>TG -&gt; CA</td>\n",
       "      <td>Substitution</td>\n",
       "      <td>Substitution</td>\n",
       "      <td>100.0%</td>\n",
       "      <td>31</td>\n",
       "      <td>25821</td>\n",
       "      <td>25822</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>C -&gt; T</td>\n",
       "      <td>Substitution</td>\n",
       "      <td>SNP (transition)</td>\n",
       "      <td>100.0%</td>\n",
       "      <td>31</td>\n",
       "      <td>25817</td>\n",
       "      <td>25817</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>T -&gt; A</td>\n",
       "      <td>None</td>\n",
       "      <td>SNP (transversion)</td>\n",
       "      <td>96.0%</td>\n",
       "      <td>32</td>\n",
       "      <td>25804</td>\n",
       "      <td>25804</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>G -&gt; A</td>\n",
       "      <td>None</td>\n",
       "      <td>SNP (transition)</td>\n",
       "      <td>96.2%</td>\n",
       "      <td>31</td>\n",
       "      <td>25801</td>\n",
       "      <td>25801</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>S -&gt; C</td>\n",
       "      <td>Substitution</td>\n",
       "      <td>SNP</td>\n",
       "      <td>96.2%</td>\n",
       "      <td>31</td>\n",
       "      <td>25799</td>\n",
       "      <td>25799</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>Y -&gt; T</td>\n",
       "      <td>Substitution</td>\n",
       "      <td>SNP</td>\n",
       "      <td>100.0%</td>\n",
       "      <td>30</td>\n",
       "      <td>25794</td>\n",
       "      <td>25794</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>IRA2 CDS</th>\n",
       "      <th>N1</th>\n",
       "      <th>0</th>\n",
       "      <td>(G)3 -&gt; (G)4</td>\n",
       "      <td>Frame Shift</td>\n",
       "      <td>Insertion (tandem repeat)</td>\n",
       "      <td>75.7%</td>\n",
       "      <td>27</td>\n",
       "      <td>177352</td>\n",
       "      <td>177351</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MDM1 CDS</th>\n",
       "      <th>N1</th>\n",
       "      <th>3</th>\n",
       "      <td>A -&gt; T</td>\n",
       "      <td>Substitution</td>\n",
       "      <td>SNP (transversion)</td>\n",
       "      <td>50.8%</td>\n",
       "      <td>20</td>\n",
       "      <td>62316</td>\n",
       "      <td>62316</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SAN1 CDS</th>\n",
       "      <th>N1</th>\n",
       "      <th>0</th>\n",
       "      <td>C -&gt; T</td>\n",
       "      <td>Substitution</td>\n",
       "      <td>SNP (transition)</td>\n",
       "      <td>81.2%</td>\n",
       "      <td>30</td>\n",
       "      <td>742965</td>\n",
       "      <td>742965</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"18\" valign=\"top\">NUP133 CDS</th>\n",
       "      <th rowspan=\"18\" valign=\"top\">N1</th>\n",
       "      <th>0</th>\n",
       "      <td>G -&gt; T</td>\n",
       "      <td>None</td>\n",
       "      <td>SNP (transversion)</td>\n",
       "      <td>50.3%</td>\n",
       "      <td>31</td>\n",
       "      <td>593290</td>\n",
       "      <td>593290</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GC -&gt; AT</td>\n",
       "      <td>Substitution</td>\n",
       "      <td>Substitution</td>\n",
       "      <td>50.0%</td>\n",
       "      <td>27</td>\n",
       "      <td>593286</td>\n",
       "      <td>593287</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GTT -&gt; TTA</td>\n",
       "      <td>Substitution</td>\n",
       "      <td>Substitution</td>\n",
       "      <td>49.4% -&gt; 50.0%</td>\n",
       "      <td>32</td>\n",
       "      <td>593282</td>\n",
       "      <td>593284</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>AAA -&gt; CTT</td>\n",
       "      <td>Substitution</td>\n",
       "      <td>Substitution</td>\n",
       "      <td>51.2% -&gt; 51.5%</td>\n",
       "      <td>32</td>\n",
       "      <td>593278</td>\n",
       "      <td>593280</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>TAA -&gt; ACT</td>\n",
       "      <td>Extension</td>\n",
       "      <td>Substitution</td>\n",
       "      <td>51.5% -&gt; 52.1%</td>\n",
       "      <td>31</td>\n",
       "      <td>593273</td>\n",
       "      <td>593275</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>T -&gt; A</td>\n",
       "      <td>Substitution</td>\n",
       "      <td>SNP (transversion)</td>\n",
       "      <td>54.2%</td>\n",
       "      <td>32</td>\n",
       "      <td>593271</td>\n",
       "      <td>593271</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>CT -&gt; TC</td>\n",
       "      <td>Substitution</td>\n",
       "      <td>Substitution</td>\n",
       "      <td>54.2% -&gt; 55.1%</td>\n",
       "      <td>32</td>\n",
       "      <td>593267</td>\n",
       "      <td>593268</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>G -&gt; C</td>\n",
       "      <td>Substitution</td>\n",
       "      <td>SNP (transversion)</td>\n",
       "      <td>57.4%</td>\n",
       "      <td>33</td>\n",
       "      <td>593264</td>\n",
       "      <td>593264</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>AG -&gt; TC</td>\n",
       "      <td>Substitution</td>\n",
       "      <td>Substitution</td>\n",
       "      <td>57.1% -&gt; 57.6%</td>\n",
       "      <td>31</td>\n",
       "      <td>593260</td>\n",
       "      <td>593261</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>ATT -&gt; GAG</td>\n",
       "      <td>Substitution</td>\n",
       "      <td>Substitution</td>\n",
       "      <td>55.8% -&gt; 56.1%</td>\n",
       "      <td>31</td>\n",
       "      <td>593255</td>\n",
       "      <td>593257</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>TA -&gt; AC</td>\n",
       "      <td>Substitution</td>\n",
       "      <td>Substitution</td>\n",
       "      <td>58.5% -&gt; 59.5%</td>\n",
       "      <td>33</td>\n",
       "      <td>593252</td>\n",
       "      <td>593253</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>TT -&gt; GC</td>\n",
       "      <td>Substitution</td>\n",
       "      <td>Substitution</td>\n",
       "      <td>60.2% -&gt; 60.7%</td>\n",
       "      <td>31</td>\n",
       "      <td>593249</td>\n",
       "      <td>593250</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>TGG -&gt; GGA</td>\n",
       "      <td>Substitution</td>\n",
       "      <td>Substitution</td>\n",
       "      <td>61.5% -&gt; 61.8%</td>\n",
       "      <td>31</td>\n",
       "      <td>593243</td>\n",
       "      <td>593245</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>G -&gt; T</td>\n",
       "      <td>Substitution</td>\n",
       "      <td>SNP (transversion)</td>\n",
       "      <td>64.6%</td>\n",
       "      <td>30</td>\n",
       "      <td>593240</td>\n",
       "      <td>593240</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>T -&gt; A</td>\n",
       "      <td>Substitution</td>\n",
       "      <td>SNP (transversion)</td>\n",
       "      <td>67.2%</td>\n",
       "      <td>33</td>\n",
       "      <td>593232</td>\n",
       "      <td>593232</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>A -&gt; G</td>\n",
       "      <td>Substitution</td>\n",
       "      <td>SNP (transition)</td>\n",
       "      <td>70.4%</td>\n",
       "      <td>33</td>\n",
       "      <td>593225</td>\n",
       "      <td>593225</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>K -&gt; G</td>\n",
       "      <td>Substitution</td>\n",
       "      <td>SNP</td>\n",
       "      <td>71.8%</td>\n",
       "      <td>32</td>\n",
       "      <td>593222</td>\n",
       "      <td>593222</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>+G</td>\n",
       "      <td>Frame Shift</td>\n",
       "      <td>Insertion</td>\n",
       "      <td>72.8%</td>\n",
       "      <td>32</td>\n",
       "      <td>593217</td>\n",
       "      <td>593216</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">RCY1 CDS</th>\n",
       "      <th rowspan=\"2\" valign=\"top\">N1</th>\n",
       "      <th>18</th>\n",
       "      <td>W -&gt; A</td>\n",
       "      <td>Substitution</td>\n",
       "      <td>SNP</td>\n",
       "      <td>61.5%</td>\n",
       "      <td>30</td>\n",
       "      <td>50664</td>\n",
       "      <td>50664</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>T -&gt; G</td>\n",
       "      <td>None</td>\n",
       "      <td>SNP (transversion)</td>\n",
       "      <td>54.7%</td>\n",
       "      <td>23</td>\n",
       "      <td>50657</td>\n",
       "      <td>50657</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>RSC30 CDS</th>\n",
       "      <th>N2</th>\n",
       "      <th>10</th>\n",
       "      <td>C -&gt; T</td>\n",
       "      <td>Substitution</td>\n",
       "      <td>SNP (transition)</td>\n",
       "      <td>50.2%</td>\n",
       "      <td>24</td>\n",
       "      <td>216122</td>\n",
       "      <td>216122</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MTL1 CDS</th>\n",
       "      <th>N2</th>\n",
       "      <th>0</th>\n",
       "      <td>M -&gt; C</td>\n",
       "      <td>None</td>\n",
       "      <td>SNP</td>\n",
       "      <td>55.4%</td>\n",
       "      <td>29</td>\n",
       "      <td>530038</td>\n",
       "      <td>530038</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>WHI2 CDS</th>\n",
       "      <th>N2</th>\n",
       "      <th>2</th>\n",
       "      <td>C -&gt; T</td>\n",
       "      <td>Truncation</td>\n",
       "      <td>SNP (transition)</td>\n",
       "      <td>58.4%</td>\n",
       "      <td>33</td>\n",
       "      <td>411549</td>\n",
       "      <td>411549</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">FLO11 CDS</th>\n",
       "      <th rowspan=\"2\" valign=\"top\">N2</th>\n",
       "      <th>0</th>\n",
       "      <td>A -&gt; T</td>\n",
       "      <td>None</td>\n",
       "      <td>SNP (transversion)</td>\n",
       "      <td>50.0%</td>\n",
       "      <td>15</td>\n",
       "      <td>392099</td>\n",
       "      <td>392099</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>G -&gt; A</td>\n",
       "      <td>Substitution</td>\n",
       "      <td>SNP (transition)</td>\n",
       "      <td>60.3%</td>\n",
       "      <td>30</td>\n",
       "      <td>391197</td>\n",
       "      <td>391197</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>hypothetical protein CDS</th>\n",
       "      <th>N2</th>\n",
       "      <th>0</th>\n",
       "      <td>G -&gt; A</td>\n",
       "      <td>None</td>\n",
       "      <td>SNP (transition)</td>\n",
       "      <td>95.2%</td>\n",
       "      <td>32</td>\n",
       "      <td>623079</td>\n",
       "      <td>623079</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"9\" valign=\"top\">FLO9 CDS</th>\n",
       "      <th rowspan=\"9\" valign=\"top\">N2</th>\n",
       "      <th>1</th>\n",
       "      <td>C -&gt; T</td>\n",
       "      <td>Substitution</td>\n",
       "      <td>SNP (transition)</td>\n",
       "      <td>94.7%</td>\n",
       "      <td>20</td>\n",
       "      <td>26492</td>\n",
       "      <td>26492</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>G -&gt; C</td>\n",
       "      <td>Substitution</td>\n",
       "      <td>SNP (transversion)</td>\n",
       "      <td>97.3%</td>\n",
       "      <td>22</td>\n",
       "      <td>26474</td>\n",
       "      <td>26474</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>CA -&gt; TG</td>\n",
       "      <td>Substitution</td>\n",
       "      <td>Substitution</td>\n",
       "      <td>100.0%</td>\n",
       "      <td>22</td>\n",
       "      <td>26469</td>\n",
       "      <td>26470</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>C -&gt; T</td>\n",
       "      <td>Substitution</td>\n",
       "      <td>SNP (transition)</td>\n",
       "      <td>100.0%</td>\n",
       "      <td>21</td>\n",
       "      <td>26464</td>\n",
       "      <td>26464</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>T -&gt; C</td>\n",
       "      <td>None</td>\n",
       "      <td>SNP (transition)</td>\n",
       "      <td>91.2%</td>\n",
       "      <td>22</td>\n",
       "      <td>26458</td>\n",
       "      <td>26458</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>A -&gt; G</td>\n",
       "      <td>None</td>\n",
       "      <td>SNP (transition)</td>\n",
       "      <td>97.1%</td>\n",
       "      <td>20</td>\n",
       "      <td>26455</td>\n",
       "      <td>26455</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>T -&gt; G</td>\n",
       "      <td>Substitution</td>\n",
       "      <td>SNP (transversion)</td>\n",
       "      <td>100.0%</td>\n",
       "      <td>20</td>\n",
       "      <td>26444</td>\n",
       "      <td>26444</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>GT -&gt; AC</td>\n",
       "      <td>Substitution</td>\n",
       "      <td>Substitution</td>\n",
       "      <td>96.3%</td>\n",
       "      <td>24</td>\n",
       "      <td>26440</td>\n",
       "      <td>26441</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>G -&gt; C</td>\n",
       "      <td>Substitution</td>\n",
       "      <td>SNP (transversion)</td>\n",
       "      <td>80.5%</td>\n",
       "      <td>25</td>\n",
       "      <td>25713</td>\n",
       "      <td>25713</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HPF1 CDS</th>\n",
       "      <th>N2</th>\n",
       "      <th>3</th>\n",
       "      <td>K -&gt; T</td>\n",
       "      <td>Substitution</td>\n",
       "      <td>SNP</td>\n",
       "      <td>64.7%</td>\n",
       "      <td>26</td>\n",
       "      <td>31133</td>\n",
       "      <td>31133</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MDM1 CDS</th>\n",
       "      <th>N2</th>\n",
       "      <th>0</th>\n",
       "      <td>A -&gt; T</td>\n",
       "      <td>Substitution</td>\n",
       "      <td>SNP (transversion)</td>\n",
       "      <td>50.0%</td>\n",
       "      <td>24</td>\n",
       "      <td>62316</td>\n",
       "      <td>62316</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"7\" valign=\"top\">NUP133 CDS</th>\n",
       "      <th rowspan=\"7\" valign=\"top\">N2</th>\n",
       "      <th>0</th>\n",
       "      <td>C -&gt; T</td>\n",
       "      <td>None</td>\n",
       "      <td>SNP (transition)</td>\n",
       "      <td>51.1%</td>\n",
       "      <td>32</td>\n",
       "      <td>593251</td>\n",
       "      <td>593251</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>T -&gt; G</td>\n",
       "      <td>Substitution</td>\n",
       "      <td>SNP (transversion)</td>\n",
       "      <td>51.1%</td>\n",
       "      <td>28</td>\n",
       "      <td>593243</td>\n",
       "      <td>593243</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>G -&gt; T</td>\n",
       "      <td>Substitution</td>\n",
       "      <td>SNP (transversion)</td>\n",
       "      <td>54.3%</td>\n",
       "      <td>28</td>\n",
       "      <td>593240</td>\n",
       "      <td>593240</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>T -&gt; A</td>\n",
       "      <td>Substitution</td>\n",
       "      <td>SNP (transversion)</td>\n",
       "      <td>50.5%</td>\n",
       "      <td>33</td>\n",
       "      <td>593232</td>\n",
       "      <td>593232</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>A -&gt; G</td>\n",
       "      <td>Substitution</td>\n",
       "      <td>SNP (transition)</td>\n",
       "      <td>54.3%</td>\n",
       "      <td>30</td>\n",
       "      <td>593225</td>\n",
       "      <td>593225</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>K -&gt; G</td>\n",
       "      <td>Substitution</td>\n",
       "      <td>SNP</td>\n",
       "      <td>56.8%</td>\n",
       "      <td>31</td>\n",
       "      <td>593222</td>\n",
       "      <td>593222</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>+G</td>\n",
       "      <td>Frame Shift</td>\n",
       "      <td>Insertion</td>\n",
       "      <td>54.1%</td>\n",
       "      <td>30</td>\n",
       "      <td>593217</td>\n",
       "      <td>593216</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">MTL1 CDS</th>\n",
       "      <th rowspan=\"2\" valign=\"top\">N3</th>\n",
       "      <th>0</th>\n",
       "      <td>M -&gt; C</td>\n",
       "      <td>None</td>\n",
       "      <td>SNP</td>\n",
       "      <td>58.2%</td>\n",
       "      <td>26</td>\n",
       "      <td>530038</td>\n",
       "      <td>530038</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>C -&gt; A</td>\n",
       "      <td>None</td>\n",
       "      <td>SNP (transversion)</td>\n",
       "      <td>51.8%</td>\n",
       "      <td>27</td>\n",
       "      <td>530020</td>\n",
       "      <td>530020</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"10\" valign=\"top\">SWI1 CDS</th>\n",
       "      <th rowspan=\"10\" valign=\"top\">N3</th>\n",
       "      <th>0</th>\n",
       "      <td>W -&gt; A</td>\n",
       "      <td>Truncation</td>\n",
       "      <td>SNP</td>\n",
       "      <td>92.4%</td>\n",
       "      <td>30</td>\n",
       "      <td>521046</td>\n",
       "      <td>521046</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>M -&gt; A</td>\n",
       "      <td>Truncation</td>\n",
       "      <td>SNP</td>\n",
       "      <td>99.0%</td>\n",
       "      <td>31</td>\n",
       "      <td>521039</td>\n",
       "      <td>521039</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>M -&gt; A</td>\n",
       "      <td>Truncation</td>\n",
       "      <td>SNP</td>\n",
       "      <td>99.0%</td>\n",
       "      <td>31</td>\n",
       "      <td>521036</td>\n",
       "      <td>521036</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>W -&gt; A</td>\n",
       "      <td>Truncation</td>\n",
       "      <td>SNP</td>\n",
       "      <td>98.1%</td>\n",
       "      <td>31</td>\n",
       "      <td>521034</td>\n",
       "      <td>521034</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>W -&gt; A</td>\n",
       "      <td>Truncation</td>\n",
       "      <td>SNP</td>\n",
       "      <td>98.1%</td>\n",
       "      <td>31</td>\n",
       "      <td>521031</td>\n",
       "      <td>521031</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>W -&gt; A</td>\n",
       "      <td>Truncation</td>\n",
       "      <td>SNP</td>\n",
       "      <td>99.0%</td>\n",
       "      <td>31</td>\n",
       "      <td>521027</td>\n",
       "      <td>521027</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>W -&gt; A</td>\n",
       "      <td>Substitution</td>\n",
       "      <td>SNP</td>\n",
       "      <td>99.0%</td>\n",
       "      <td>31</td>\n",
       "      <td>521024</td>\n",
       "      <td>521024</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>MK -&gt; AT</td>\n",
       "      <td>Substitution</td>\n",
       "      <td>Substitution</td>\n",
       "      <td>93.1%</td>\n",
       "      <td>28</td>\n",
       "      <td>521018</td>\n",
       "      <td>521019</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>WW -&gt; TT</td>\n",
       "      <td>Substitution</td>\n",
       "      <td>Substitution</td>\n",
       "      <td>94.8% -&gt; 94.9%</td>\n",
       "      <td>29</td>\n",
       "      <td>521014</td>\n",
       "      <td>521015</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>SAW -&gt; GAT</td>\n",
       "      <td>Substitution</td>\n",
       "      <td>Substitution</td>\n",
       "      <td>90.5% -&gt; 90.6%</td>\n",
       "      <td>26</td>\n",
       "      <td>521008</td>\n",
       "      <td>521010</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">FLO11 CDS</th>\n",
       "      <th rowspan=\"2\" valign=\"top\">N3</th>\n",
       "      <th>0</th>\n",
       "      <td>C -&gt; G</td>\n",
       "      <td>Substitution</td>\n",
       "      <td>SNP (transversion)</td>\n",
       "      <td>53.3%</td>\n",
       "      <td>11</td>\n",
       "      <td>391590</td>\n",
       "      <td>391590</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>G -&gt; A</td>\n",
       "      <td>Substitution</td>\n",
       "      <td>SNP (transition)</td>\n",
       "      <td>57.5%</td>\n",
       "      <td>27</td>\n",
       "      <td>391197</td>\n",
       "      <td>391197</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>hypothetical protein CDS</th>\n",
       "      <th>N3</th>\n",
       "      <th>1</th>\n",
       "      <td>G -&gt; A</td>\n",
       "      <td>None</td>\n",
       "      <td>SNP (transition)</td>\n",
       "      <td>98.8%</td>\n",
       "      <td>31</td>\n",
       "      <td>623079</td>\n",
       "      <td>623079</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>FLO9 CDS</th>\n",
       "      <th>N3</th>\n",
       "      <th>1</th>\n",
       "      <td>A -&gt; C</td>\n",
       "      <td>Substitution</td>\n",
       "      <td>SNP (transversion)</td>\n",
       "      <td>86.2%</td>\n",
       "      <td>31</td>\n",
       "      <td>25848</td>\n",
       "      <td>25848</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HPF1 CDS</th>\n",
       "      <th>N3</th>\n",
       "      <th>1</th>\n",
       "      <td>K -&gt; T</td>\n",
       "      <td>Substitution</td>\n",
       "      <td>SNP</td>\n",
       "      <td>51.2%</td>\n",
       "      <td>25</td>\n",
       "      <td>31133</td>\n",
       "      <td>31133</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"4\" valign=\"top\">NUP133 CDS</th>\n",
       "      <th rowspan=\"4\" valign=\"top\">N3</th>\n",
       "      <th>0</th>\n",
       "      <td>G -&gt; T</td>\n",
       "      <td>Substitution</td>\n",
       "      <td>SNP (transversion)</td>\n",
       "      <td>56.5%</td>\n",
       "      <td>28</td>\n",
       "      <td>593240</td>\n",
       "      <td>593240</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>T -&gt; A</td>\n",
       "      <td>Substitution</td>\n",
       "      <td>SNP (transversion)</td>\n",
       "      <td>56.4%</td>\n",
       "      <td>30</td>\n",
       "      <td>593232</td>\n",
       "      <td>593232</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>K -&gt; G</td>\n",
       "      <td>Substitution</td>\n",
       "      <td>SNP</td>\n",
       "      <td>60.6%</td>\n",
       "      <td>29</td>\n",
       "      <td>593222</td>\n",
       "      <td>593222</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>+G</td>\n",
       "      <td>Frame Shift</td>\n",
       "      <td>Insertion</td>\n",
       "      <td>60.2%</td>\n",
       "      <td>30</td>\n",
       "      <td>593217</td>\n",
       "      <td>593216</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>RCY1 CDS</th>\n",
       "      <th>N3</th>\n",
       "      <th>6</th>\n",
       "      <td>W -&gt; A</td>\n",
       "      <td>Substitution</td>\n",
       "      <td>SNP</td>\n",
       "      <td>54.1%</td>\n",
       "      <td>24</td>\n",
       "      <td>50664</td>\n",
       "      <td>50664</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                              Change Protein Effect  \\\n",
       "Gene                     Sample Number                                \n",
       "MTL1 CDS                 N1     0             M -> C           None   \n",
       "DAN4 CDS                 N1     4             T -> G   Substitution   \n",
       "                                5             T -> G   Substitution   \n",
       "                                6             C -> G   Substitution   \n",
       "SWI1 CDS                 N1     8             W -> A     Truncation   \n",
       "                                9             M -> A     Truncation   \n",
       "                                10            M -> A     Truncation   \n",
       "                                11            W -> A     Truncation   \n",
       "                                12            W -> A     Truncation   \n",
       "                                13            W -> A     Truncation   \n",
       "                                14            W -> A   Substitution   \n",
       "                                15          MK -> AT   Substitution   \n",
       "                                16          WW -> TT   Substitution   \n",
       "                                17        SAW -> GAT   Substitution   \n",
       "FLO11 CDS                N1     0             C -> G   Substitution   \n",
       "                                1             C -> G   Substitution   \n",
       "                                2             G -> T           None   \n",
       "                                3             G -> A   Substitution   \n",
       "hypothetical protein CDS N1     6             T -> G           None   \n",
       "                                7           CA -> TG   Substitution   \n",
       "                                8       AACA -> TCAT   Substitution   \n",
       "                                9           GT -> CA   Substitution   \n",
       "                                10      CTAA -> ATTT   Substitution   \n",
       "FLO9 CDS                 N1     2             G -> A           None   \n",
       "                                3             C -> T   Substitution   \n",
       "                                4             T -> A           None   \n",
       "                                5             G -> A           None   \n",
       "                                6           AG -> GC   Substitution   \n",
       "                                7         GCA -> ATG   Substitution   \n",
       "                                8             C -> T   Substitution   \n",
       "                                9           TG -> CA   Substitution   \n",
       "                                10            C -> T   Substitution   \n",
       "                                11            T -> A           None   \n",
       "                                12            G -> A           None   \n",
       "                                13            S -> C   Substitution   \n",
       "                                14            Y -> T   Substitution   \n",
       "IRA2 CDS                 N1     0       (G)3 -> (G)4    Frame Shift   \n",
       "MDM1 CDS                 N1     3             A -> T   Substitution   \n",
       "SAN1 CDS                 N1     0             C -> T   Substitution   \n",
       "NUP133 CDS               N1     0             G -> T           None   \n",
       "                                1           GC -> AT   Substitution   \n",
       "                                2         GTT -> TTA   Substitution   \n",
       "                                3         AAA -> CTT   Substitution   \n",
       "                                4         TAA -> ACT      Extension   \n",
       "                                5             T -> A   Substitution   \n",
       "                                6           CT -> TC   Substitution   \n",
       "                                7             G -> C   Substitution   \n",
       "                                8           AG -> TC   Substitution   \n",
       "                                9         ATT -> GAG   Substitution   \n",
       "                                10          TA -> AC   Substitution   \n",
       "                                11          TT -> GC   Substitution   \n",
       "                                12        TGG -> GGA   Substitution   \n",
       "                                13            G -> T   Substitution   \n",
       "                                14            T -> A   Substitution   \n",
       "                                15            A -> G   Substitution   \n",
       "                                16            K -> G   Substitution   \n",
       "                                17                +G    Frame Shift   \n",
       "RCY1 CDS                 N1     18            W -> A   Substitution   \n",
       "                                19            T -> G           None   \n",
       "RSC30 CDS                N2     10            C -> T   Substitution   \n",
       "MTL1 CDS                 N2     0             M -> C           None   \n",
       "WHI2 CDS                 N2     2             C -> T     Truncation   \n",
       "FLO11 CDS                N2     0             A -> T           None   \n",
       "                                1             G -> A   Substitution   \n",
       "hypothetical protein CDS N2     0             G -> A           None   \n",
       "FLO9 CDS                 N2     1             C -> T   Substitution   \n",
       "                                2             G -> C   Substitution   \n",
       "                                3           CA -> TG   Substitution   \n",
       "                                4             C -> T   Substitution   \n",
       "                                5             T -> C           None   \n",
       "                                6             A -> G           None   \n",
       "                                7             T -> G   Substitution   \n",
       "                                8           GT -> AC   Substitution   \n",
       "                                9             G -> C   Substitution   \n",
       "HPF1 CDS                 N2     3             K -> T   Substitution   \n",
       "MDM1 CDS                 N2     0             A -> T   Substitution   \n",
       "NUP133 CDS               N2     0             C -> T           None   \n",
       "                                1             T -> G   Substitution   \n",
       "                                2             G -> T   Substitution   \n",
       "                                3             T -> A   Substitution   \n",
       "                                4             A -> G   Substitution   \n",
       "                                5             K -> G   Substitution   \n",
       "                                6                 +G    Frame Shift   \n",
       "MTL1 CDS                 N3     0             M -> C           None   \n",
       "                                1             C -> A           None   \n",
       "SWI1 CDS                 N3     0             W -> A     Truncation   \n",
       "                                1             M -> A     Truncation   \n",
       "                                2             M -> A     Truncation   \n",
       "                                3             W -> A     Truncation   \n",
       "                                4             W -> A     Truncation   \n",
       "                                5             W -> A     Truncation   \n",
       "                                6             W -> A   Substitution   \n",
       "                                7           MK -> AT   Substitution   \n",
       "                                8           WW -> TT   Substitution   \n",
       "                                9         SAW -> GAT   Substitution   \n",
       "FLO11 CDS                N3     0             C -> G   Substitution   \n",
       "                                1             G -> A   Substitution   \n",
       "hypothetical protein CDS N3     1             G -> A           None   \n",
       "FLO9 CDS                 N3     1             A -> C   Substitution   \n",
       "HPF1 CDS                 N3     1             K -> T   Substitution   \n",
       "NUP133 CDS               N3     0             G -> T   Substitution   \n",
       "                                1             T -> A   Substitution   \n",
       "                                2             K -> G   Substitution   \n",
       "                                3                 +G    Frame Shift   \n",
       "RCY1 CDS                 N3     6             W -> A   Substitution   \n",
       "\n",
       "                                                Polymorphism Type  \\\n",
       "Gene                     Sample Number                              \n",
       "MTL1 CDS                 N1     0                             SNP   \n",
       "DAN4 CDS                 N1     4              SNP (transversion)   \n",
       "                                5              SNP (transversion)   \n",
       "                                6              SNP (transversion)   \n",
       "SWI1 CDS                 N1     8                             SNP   \n",
       "                                9                             SNP   \n",
       "                                10                            SNP   \n",
       "                                11                            SNP   \n",
       "                                12                            SNP   \n",
       "                                13                            SNP   \n",
       "                                14                            SNP   \n",
       "                                15                   Substitution   \n",
       "                                16                   Substitution   \n",
       "                                17                   Substitution   \n",
       "FLO11 CDS                N1     0              SNP (transversion)   \n",
       "                                1              SNP (transversion)   \n",
       "                                2              SNP (transversion)   \n",
       "                                3                SNP (transition)   \n",
       "hypothetical protein CDS N1     6              SNP (transversion)   \n",
       "                                7                    Substitution   \n",
       "                                8                    Substitution   \n",
       "                                9                    Substitution   \n",
       "                                10                   Substitution   \n",
       "FLO9 CDS                 N1     2                SNP (transition)   \n",
       "                                3                SNP (transition)   \n",
       "                                4              SNP (transversion)   \n",
       "                                5                SNP (transition)   \n",
       "                                6                    Substitution   \n",
       "                                7                    Substitution   \n",
       "                                8                SNP (transition)   \n",
       "                                9                    Substitution   \n",
       "                                10               SNP (transition)   \n",
       "                                11             SNP (transversion)   \n",
       "                                12               SNP (transition)   \n",
       "                                13                            SNP   \n",
       "                                14                            SNP   \n",
       "IRA2 CDS                 N1     0       Insertion (tandem repeat)   \n",
       "MDM1 CDS                 N1     3              SNP (transversion)   \n",
       "SAN1 CDS                 N1     0                SNP (transition)   \n",
       "NUP133 CDS               N1     0              SNP (transversion)   \n",
       "                                1                    Substitution   \n",
       "                                2                    Substitution   \n",
       "                                3                    Substitution   \n",
       "                                4                    Substitution   \n",
       "                                5              SNP (transversion)   \n",
       "                                6                    Substitution   \n",
       "                                7              SNP (transversion)   \n",
       "                                8                    Substitution   \n",
       "                                9                    Substitution   \n",
       "                                10                   Substitution   \n",
       "                                11                   Substitution   \n",
       "                                12                   Substitution   \n",
       "                                13             SNP (transversion)   \n",
       "                                14             SNP (transversion)   \n",
       "                                15               SNP (transition)   \n",
       "                                16                            SNP   \n",
       "                                17                      Insertion   \n",
       "RCY1 CDS                 N1     18                            SNP   \n",
       "                                19             SNP (transversion)   \n",
       "RSC30 CDS                N2     10               SNP (transition)   \n",
       "MTL1 CDS                 N2     0                             SNP   \n",
       "WHI2 CDS                 N2     2                SNP (transition)   \n",
       "FLO11 CDS                N2     0              SNP (transversion)   \n",
       "                                1                SNP (transition)   \n",
       "hypothetical protein CDS N2     0                SNP (transition)   \n",
       "FLO9 CDS                 N2     1                SNP (transition)   \n",
       "                                2              SNP (transversion)   \n",
       "                                3                    Substitution   \n",
       "                                4                SNP (transition)   \n",
       "                                5                SNP (transition)   \n",
       "                                6                SNP (transition)   \n",
       "                                7              SNP (transversion)   \n",
       "                                8                    Substitution   \n",
       "                                9              SNP (transversion)   \n",
       "HPF1 CDS                 N2     3                             SNP   \n",
       "MDM1 CDS                 N2     0              SNP (transversion)   \n",
       "NUP133 CDS               N2     0                SNP (transition)   \n",
       "                                1              SNP (transversion)   \n",
       "                                2              SNP (transversion)   \n",
       "                                3              SNP (transversion)   \n",
       "                                4                SNP (transition)   \n",
       "                                5                             SNP   \n",
       "                                6                       Insertion   \n",
       "MTL1 CDS                 N3     0                             SNP   \n",
       "                                1              SNP (transversion)   \n",
       "SWI1 CDS                 N3     0                             SNP   \n",
       "                                1                             SNP   \n",
       "                                2                             SNP   \n",
       "                                3                             SNP   \n",
       "                                4                             SNP   \n",
       "                                5                             SNP   \n",
       "                                6                             SNP   \n",
       "                                7                    Substitution   \n",
       "                                8                    Substitution   \n",
       "                                9                    Substitution   \n",
       "FLO11 CDS                N3     0              SNP (transversion)   \n",
       "                                1                SNP (transition)   \n",
       "hypothetical protein CDS N3     1                SNP (transition)   \n",
       "FLO9 CDS                 N3     1              SNP (transversion)   \n",
       "HPF1 CDS                 N3     1                             SNP   \n",
       "NUP133 CDS               N3     0              SNP (transversion)   \n",
       "                                1              SNP (transversion)   \n",
       "                                2                             SNP   \n",
       "                                3                       Insertion   \n",
       "RCY1 CDS                 N3     6                             SNP   \n",
       "\n",
       "                                       Variant Frequency  Average Quality  \\\n",
       "Gene                     Sample Number                                      \n",
       "MTL1 CDS                 N1     0                  58.0%               26   \n",
       "DAN4 CDS                 N1     4                  54.3%               15   \n",
       "                                5                  63.8%               17   \n",
       "                                6                  59.1%               16   \n",
       "SWI1 CDS                 N1     8                  97.0%               32   \n",
       "                                9                  97.1%               33   \n",
       "                                10                100.0%               34   \n",
       "                                11                 99.0%               34   \n",
       "                                12                 99.0%               32   \n",
       "                                13                 99.0%               32   \n",
       "                                14                 97.0%               32   \n",
       "                                15                 96.9%               31   \n",
       "                                16                 98.0%               31   \n",
       "                                17        92.6% -> 92.8%               29   \n",
       "FLO11 CDS                N1     0                  53.8%               12   \n",
       "                                1                  50.9%               12   \n",
       "                                2                  53.7%               21   \n",
       "                                3                  66.4%               28   \n",
       "hypothetical protein CDS N1     6                  73.3%               26   \n",
       "                                7                  59.6%               24   \n",
       "                                8                  59.6%               24   \n",
       "                                9                  58.3%               24   \n",
       "                                10        50.0% -> 52.2%               23   \n",
       "FLO9 CDS                 N1     2                  97.6%               24   \n",
       "                                3                  92.1%               22   \n",
       "                                4                  97.1%               24   \n",
       "                                5                  97.1%               22   \n",
       "                                6         96.9% -> 97.0%               23   \n",
       "                                7                  96.8%               24   \n",
       "                                8                 100.0%               23   \n",
       "                                9                 100.0%               31   \n",
       "                                10                100.0%               31   \n",
       "                                11                 96.0%               32   \n",
       "                                12                 96.2%               31   \n",
       "                                13                 96.2%               31   \n",
       "                                14                100.0%               30   \n",
       "IRA2 CDS                 N1     0                  75.7%               27   \n",
       "MDM1 CDS                 N1     3                  50.8%               20   \n",
       "SAN1 CDS                 N1     0                  81.2%               30   \n",
       "NUP133 CDS               N1     0                  50.3%               31   \n",
       "                                1                  50.0%               27   \n",
       "                                2         49.4% -> 50.0%               32   \n",
       "                                3         51.2% -> 51.5%               32   \n",
       "                                4         51.5% -> 52.1%               31   \n",
       "                                5                  54.2%               32   \n",
       "                                6         54.2% -> 55.1%               32   \n",
       "                                7                  57.4%               33   \n",
       "                                8         57.1% -> 57.6%               31   \n",
       "                                9         55.8% -> 56.1%               31   \n",
       "                                10        58.5% -> 59.5%               33   \n",
       "                                11        60.2% -> 60.7%               31   \n",
       "                                12        61.5% -> 61.8%               31   \n",
       "                                13                 64.6%               30   \n",
       "                                14                 67.2%               33   \n",
       "                                15                 70.4%               33   \n",
       "                                16                 71.8%               32   \n",
       "                                17                 72.8%               32   \n",
       "RCY1 CDS                 N1     18                 61.5%               30   \n",
       "                                19                 54.7%               23   \n",
       "RSC30 CDS                N2     10                 50.2%               24   \n",
       "MTL1 CDS                 N2     0                  55.4%               29   \n",
       "WHI2 CDS                 N2     2                  58.4%               33   \n",
       "FLO11 CDS                N2     0                  50.0%               15   \n",
       "                                1                  60.3%               30   \n",
       "hypothetical protein CDS N2     0                  95.2%               32   \n",
       "FLO9 CDS                 N2     1                  94.7%               20   \n",
       "                                2                  97.3%               22   \n",
       "                                3                 100.0%               22   \n",
       "                                4                 100.0%               21   \n",
       "                                5                  91.2%               22   \n",
       "                                6                  97.1%               20   \n",
       "                                7                 100.0%               20   \n",
       "                                8                  96.3%               24   \n",
       "                                9                  80.5%               25   \n",
       "HPF1 CDS                 N2     3                  64.7%               26   \n",
       "MDM1 CDS                 N2     0                  50.0%               24   \n",
       "NUP133 CDS               N2     0                  51.1%               32   \n",
       "                                1                  51.1%               28   \n",
       "                                2                  54.3%               28   \n",
       "                                3                  50.5%               33   \n",
       "                                4                  54.3%               30   \n",
       "                                5                  56.8%               31   \n",
       "                                6                  54.1%               30   \n",
       "MTL1 CDS                 N3     0                  58.2%               26   \n",
       "                                1                  51.8%               27   \n",
       "SWI1 CDS                 N3     0                  92.4%               30   \n",
       "                                1                  99.0%               31   \n",
       "                                2                  99.0%               31   \n",
       "                                3                  98.1%               31   \n",
       "                                4                  98.1%               31   \n",
       "                                5                  99.0%               31   \n",
       "                                6                  99.0%               31   \n",
       "                                7                  93.1%               28   \n",
       "                                8         94.8% -> 94.9%               29   \n",
       "                                9         90.5% -> 90.6%               26   \n",
       "FLO11 CDS                N3     0                  53.3%               11   \n",
       "                                1                  57.5%               27   \n",
       "hypothetical protein CDS N3     1                  98.8%               31   \n",
       "FLO9 CDS                 N3     1                  86.2%               31   \n",
       "HPF1 CDS                 N3     1                  51.2%               25   \n",
       "NUP133 CDS               N3     0                  56.5%               28   \n",
       "                                1                  56.4%               30   \n",
       "                                2                  60.6%               29   \n",
       "                                3                  60.2%               30   \n",
       "RCY1 CDS                 N3     6                  54.1%               24   \n",
       "\n",
       "                                        Minimum  Maximum  \n",
       "Gene                     Sample Number                    \n",
       "MTL1 CDS                 N1     0        530038   530038  \n",
       "DAN4 CDS                 N1     4        715015   715015  \n",
       "                                5        715003   715003  \n",
       "                                6        714979   714979  \n",
       "SWI1 CDS                 N1     8        521046   521046  \n",
       "                                9        521039   521039  \n",
       "                                10       521036   521036  \n",
       "                                11       521034   521034  \n",
       "                                12       521031   521031  \n",
       "                                13       521027   521027  \n",
       "                                14       521024   521024  \n",
       "                                15       521018   521019  \n",
       "                                16       521014   521015  \n",
       "                                17       521008   521010  \n",
       "FLO11 CDS                N1     0        392337   392337  \n",
       "                                1        392322   392322  \n",
       "                                2        392087   392087  \n",
       "                                3        391197   391197  \n",
       "hypothetical protein CDS N1     6        475850   475850  \n",
       "                                7        475842   475843  \n",
       "                                8        475837   475840  \n",
       "                                9        475834   475835  \n",
       "                                10       475827   475830  \n",
       "FLO9 CDS                 N1     2         26497    26497  \n",
       "                                3         26492    26492  \n",
       "                                4         26479    26479  \n",
       "                                5         26476    26476  \n",
       "                                6         26473    26474  \n",
       "                                7         26468    26470  \n",
       "                                8         26464    26464  \n",
       "                                9         25821    25822  \n",
       "                                10        25817    25817  \n",
       "                                11        25804    25804  \n",
       "                                12        25801    25801  \n",
       "                                13        25799    25799  \n",
       "                                14        25794    25794  \n",
       "IRA2 CDS                 N1     0        177352   177351  \n",
       "MDM1 CDS                 N1     3         62316    62316  \n",
       "SAN1 CDS                 N1     0        742965   742965  \n",
       "NUP133 CDS               N1     0        593290   593290  \n",
       "                                1        593286   593287  \n",
       "                                2        593282   593284  \n",
       "                                3        593278   593280  \n",
       "                                4        593273   593275  \n",
       "                                5        593271   593271  \n",
       "                                6        593267   593268  \n",
       "                                7        593264   593264  \n",
       "                                8        593260   593261  \n",
       "                                9        593255   593257  \n",
       "                                10       593252   593253  \n",
       "                                11       593249   593250  \n",
       "                                12       593243   593245  \n",
       "                                13       593240   593240  \n",
       "                                14       593232   593232  \n",
       "                                15       593225   593225  \n",
       "                                16       593222   593222  \n",
       "                                17       593217   593216  \n",
       "RCY1 CDS                 N1     18        50664    50664  \n",
       "                                19        50657    50657  \n",
       "RSC30 CDS                N2     10       216122   216122  \n",
       "MTL1 CDS                 N2     0        530038   530038  \n",
       "WHI2 CDS                 N2     2        411549   411549  \n",
       "FLO11 CDS                N2     0        392099   392099  \n",
       "                                1        391197   391197  \n",
       "hypothetical protein CDS N2     0        623079   623079  \n",
       "FLO9 CDS                 N2     1         26492    26492  \n",
       "                                2         26474    26474  \n",
       "                                3         26469    26470  \n",
       "                                4         26464    26464  \n",
       "                                5         26458    26458  \n",
       "                                6         26455    26455  \n",
       "                                7         26444    26444  \n",
       "                                8         26440    26441  \n",
       "                                9         25713    25713  \n",
       "HPF1 CDS                 N2     3         31133    31133  \n",
       "MDM1 CDS                 N2     0         62316    62316  \n",
       "NUP133 CDS               N2     0        593251   593251  \n",
       "                                1        593243   593243  \n",
       "                                2        593240   593240  \n",
       "                                3        593232   593232  \n",
       "                                4        593225   593225  \n",
       "                                5        593222   593222  \n",
       "                                6        593217   593216  \n",
       "MTL1 CDS                 N3     0        530038   530038  \n",
       "                                1        530020   530020  \n",
       "SWI1 CDS                 N3     0        521046   521046  \n",
       "                                1        521039   521039  \n",
       "                                2        521036   521036  \n",
       "                                3        521034   521034  \n",
       "                                4        521031   521031  \n",
       "                                5        521027   521027  \n",
       "                                6        521024   521024  \n",
       "                                7        521018   521019  \n",
       "                                8        521014   521015  \n",
       "                                9        521008   521010  \n",
       "FLO11 CDS                N3     0        391590   391590  \n",
       "                                1        391197   391197  \n",
       "hypothetical protein CDS N3     1        623079   623079  \n",
       "FLO9 CDS                 N3     1         25848    25848  \n",
       "HPF1 CDS                 N3     1         31133    31133  \n",
       "NUP133 CDS               N3     0        593240   593240  \n",
       "                                1        593232   593232  \n",
       "                                2        593222   593222  \n",
       "                                3        593217   593216  \n",
       "RCY1 CDS                 N3     6         50664    50664  "
      ]
     },
     "execution_count": 130,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "N1_N2_N3_df"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### let's make xls"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "writer = pd.ExcelWriter(\"{}/{}.xls\".format(xls_out_dir, sheet_name))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "for i in set(strain_gene_day_N1_CDSs +\n",
    "             strain_gene_day_N2_CDSs +\n",
    "             strain_gene_day_N3_CDSs):\n",
    "    N1_N2_N3_df.xs(i).to_excel(excel_writer=writer,\n",
    "                               sheet_name=i)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "writer.save()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### let's make sheet on gdrive"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "sh = gc.create(sheet_name)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "for i in set(strain_gene_day_N1_CDSs +\n",
    "             strain_gene_day_N2_CDSs +\n",
    "             strain_gene_day_N3_CDSs):\n",
    "    time.sleep(3)\n",
    "    wks = sh.add_worksheet(i)\n",
    "    wks.set_dataframe(N1_N2_N3_df.xs(i),\n",
    "                      (1, 1),\n",
    "                      copy_index=True,\n",
    "                      fit=True)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python [default]",
   "language": "python",
   "name": "python2"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.13"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
